### PR TITLE
fix typos in typescript documentation

### PR DESCRIPTION
**Pull request by garanews**
fix typos in typescript documentation

### CODE REVIEW

**Review by SSwiniarski — APPROVED** [review]
@garanews, looks good.

File: content/typescript/concepts/narrowing/narrowing.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ TypeScript recognizes types as being narrowed when it sees branches of JavaScrip
 
 - `if/else` conditional statements.
 - Other conditionals, like ternaries and switches.
-- Looping statments, such as `for`- and `while`-loops.
+- Looping statements, such as `for`- and `while`-loops.
 - Truthiness checks for entities that don't point to `false`-like values.
```
Comment (Dusch4593) [inline content/typescript/concepts/narrowing/narrowing.md:26]:
```suggestion
- Conditional statements like `if/else`, ternaries, and switches.
- Looping statements, such as `for`- and `while`-loops.
- Truthiness checks for entities that don't point to `false`-like values.
```

File: content/typescript/concepts/narrowing/narrowing.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ logValueIfExists('narrowing');
 
 We invoked `logValueIfExists()` with a string variable, "narrowing". Since the `value` is of type `string`, the code in the `if` statement will run and a new string will be logged to the console.
 
-## Exapmle: Narrowing with truthiness
+## Example: Narrowing with truthiness
```
Comment (Dusch4593) [inline content/typescript/concepts/narrowing/narrowing.md:47]:
```suggestion
## Example: Narrowing With Truthiness
```

For Docs entries, we try to keep all headers in "Title Case."

File: content/typescript/concepts/assignability/assignability.md [hunk 3]
```diff
@@ -15,7 +15,7 @@ Assignability is how TypeScript can determine whether a value of a particular da
 
 ## When/where assignability takes place
```
Comment (Dusch4593) [inline content/typescript/concepts/assignability/assignability.md:16]:
```suggestion
## When/Where Assignability Takes Place
```

File: content/typescript/concepts/classes/classes.md [hunk 4]
```diff
@@ -123,7 +123,7 @@ class Derived extends Base {
 
 Notice how `first`, `second`, and even the protected `third`, are all accessible within the `Derived` class definition. With `fourth` and `fifth`, however, they are both private members of `Base`. Therefore, they are not accessible inside of `Derived`.
```
Comment (Dusch4593) [inline content/typescript/concepts/classes/classes.md:124]:
```suggestion
The members `first`, `second`, and even the protected `third`, are all accessible within the `Derived` class definition. With `fourth` and `fifth`, however, they are both private members of `Base`. Therefore, they are not accessible inside of `Derived`.
```

File: content/typescript/concepts/narrowing/narrowing.md (hunk 1 above)
Comment (Dusch4593) [reply content/typescript/concepts/narrowing/narrowing.md:26]: @garanews Any chance we could implement this suggestion? The first two list items seem inherently related, hence why I recommended consolidating into one item about conditional statements.

Comment (garanews) [reply content/typescript/concepts/narrowing/narrowing.md]: @Dusch4593 was not able to commit your suggestion, had to edit by my side, should be ok now.

**Review by Dusch4593 — APPROVED** [review]
Thanks, @garanews, this seems good to merge! 🚀 Just wanted to let you know some edits I just made in 1b74694c303751d52cc23a73eb6a8645fd2d3dde:

Under `assignability.md`:
- Converted headers to "Title Case".
- Separated error messages into separate \```shell blocks.

Under `classes.md`:
- Converted error comments to mult-line.

Under `generics.md`:
- I edited the `Description` to be a bit shorter.
- Linked to existing TypeScript entries for list items in lines # 30 through 33.

**Comment by Dusch4593** [comment]
@garanews The changes should now be live:

- Assignability: https://www.codecademy.com/resources/docs/typescript/assignability
- Classes: https://www.codecademy.com/resources/docs/typescript/classes
- Functions: https://www.codecademy.com/resources/docs/typescript/functions
- Generics: https://www.codecademy.com/resources/docs/typescript/generics
- Narrowing: https://www.codecademy.com/resources/docs/typescript/narrowing

Btw, I'm still not seeing your profile credits as a contributor for these entries. If you'd like for your profile to be listed, I'm more than happy to help with making that happen. 😄

**Comment by garanews** [comment]
@Dusch4593 was not aware about contributors, for me it would be a pleasure if listed!

**Comment by Dusch4593** [comment]
@garanews Awesome 😄 Let's first confirm that your GitHub profile is listed under your Codecademy account. On Codecademy, if you select your profile image (near the top-right), and then select "Profile", and then select "Edit Profile" (towards the left), you'll be brought to the screen where you can set your GitHub profile:

Screen Shot 2022-10-12 at 12 33 47 PM

After entering your GitHub username (without an `@`), select "Save Changes". Then, log out of your Codecademy profile and re-log in through GitHub, like so:

Screen Shot 2022-10-12 at 12 36 57 PM

This should allow your profile to be listed (though it may take a minute for the update to set in). Let me know if you have any questions about these steps. 😄

**Comment by garanews** [comment]
@Dusch4593 it worked perfectly! THANKS! (I've sent you an email some day ago asking about how to add some new library etc, was thinking email is better than comments here, but if you like use github comments we can continue, or better I will open an issue and start from there, let me know!)

**Comment by Dusch4593** [comment]
Hey @garanews glad to see your profile is now appearing on the entries! 😄

Thank you for sending the email, though I would advise that the best way to continue that discussion is in the Docs repository's [Discussions tab](https://github.com/Codecademy/docs/discussions) where you can post things like questions, concerns, ideas, etc. But I'm happy to provide an answer to your questions here:

Short answer is **yes**.

The collective hope on our end is that Codecademy Docs becomes a reference for all things technical, including languages, frameworks, libraries, etc., as well as a space for everyone to foster their skills with collaborating on and contributing to projects on GitHub. We don't wish to be a replacement for other documentation resources but rather a place where learners can elevate their conceptual understanding of topics like React and feel ready enough to venture into other documentation/references.

Docs has grown a lot in the 1.5 years it's been around thus far. Initially, we only had entries that covered concepts/terms under specific programming languages and frameworks (along with a General topic for language-agnostic concepts). Since then, Docs has started to grow beyond those constraints. For example, we recently released a new topic on [cloud computing](https://www.codecademy.com/resources/docs/cloud-computing) (to coincide with Codecademy's [Foundations of Cloud Computing course](https://www.codecademy.com/learn/foundations-of-cloud-computing)) as well as a new topic called [Docs Community](https://www.codecademy.com/resources/docs/docs-community) that covers aspects of working with Docs such as this very repository, reviews, entries, etc.

**Pull request merged**